core/lb_test.go

```diff
@@ -69,6 +69,43 @@ func TestLB_Ratchet(t *testing.T) {
 	})
 }
 
+func TestLB_SessionCleanupRace(t *testing.T) {
+	// Reproduce race condition around session cleanup #1750
+
+	assert := assert.New(t)
+	lb := NewLoadBalancingTranscoder("0", newStubTranscoder).(*LoadBalancingTranscoder)
+	sess := "sess"
+	// Force create a new session
+	_, err := lb.Transcode(stubMetadata(sess, ffmpeg.P144p30fps16x9))
+	assert.Nil(err)
+	// Mark transcoder to error out
+	transcoder := lb.sessions[sess].transcoder.(*StubTranscoder)
+	transcoder.FailTranscode = true
+
+	// Send 2 jobs concurrently to trigger the stuck issue
+	wg := newWg(2)
+	errSignal := make(chan bool)
+	// Error out on Job 1
+	go func() {
+		lb.mu.Lock() // lock the LB to prevent session cleanup
+		_, err = lb.sessions[sess].Transcode(stubMetadata(sess, ffmpeg.P144p30fps16x9))
+		assert.Equal(ErrTranscode, err)
+		errSignal <- true
+		wg.Done()
+	}()
+	// Job 2 arrives when the session loop() has closed, but session isn't cleaned up yet
+	go func() {
+		sig := <-errSignal
```

Comment:
Minor: Doesn't seem like a bool is required for the channel value so you can avoid some extra checks here if the channel used an empty struct instead.

For example:

```golang
sigCh := make(chan struct{})
go func() {
    ...
    sigCh <- struct{}
    wg.Done()
}()

go func() {
    <-sigCh
    ...
    wg.Done()
}()